test/commands/user_track/generate_concepts_status_test.rb

```diff
@@ -0,0 +1,201 @@
+require "test_helper"
+
+class UserTrack::GenerateConceptsStatusTest < ActiveSupport::TestCase
+  test "generates concepts for empty track" do
+    track = create :track
+    user_track = create :user_track, track: track
+
+    assert_equal(
+      {},
+      ::UserTrack::GenerateConceptsStatus.(user_track)
+    )
+  end
+
+  test "concepts without prerequisite are :unlocked" do
+    track = create :track
+    user_track = create :user_track, track: track
+
+    create :track_concept, slug: 'basics', track: track
```

Comment:
Where possible I like avoiding setting the params on factories. This means if we want to change defaults etc, it becomes easier. 

So as a general rule I try and do this in situations where I'd only use track once or twice.

```suggestion
    user_track = create :user_track

    create :track_concept, slug: 'basics', track: user_track.track
```

---

If I was using track lots, I'd either keep it as you had it (probably 50% of the time) or do:
```ruby
    user_track = create :user_track
    track = user_track.track

    ...
```

I can't articulate when I'd choose which, but I'll think on it.